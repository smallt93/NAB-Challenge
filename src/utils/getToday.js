const weekDay = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];

export const getDayOfWeek = (date) => {
  const getDay = String(date.getDate()).padStart(2, '0');
  return `${weekDay[date.getDay()]} ${getDay}`;
};