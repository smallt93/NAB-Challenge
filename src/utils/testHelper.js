import React from 'react';
import { mount, shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from 'stylesheets/defaultTheme.styles';


export function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    ),
  });
}

export function shallowWithTheme(child) {
  return shallow(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    ),
  });
}
