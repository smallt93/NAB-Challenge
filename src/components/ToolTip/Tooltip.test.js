import React from 'react';
import { mountWithTheme as mount } from 'utils/testHelper';
import Tooltip from 'components/Tooltip';

describe('render Tooltip component', () => {
  let Component;

  const mockProps = {
    align: 'left',
    title: '',
    infoList: [],
  }
  
  beforeEach(() => {
    Component = mount(<Tooltip {...mockProps} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render', () => {
    expect(Component.length).toBe(1);
  });

  it('should match snapshot', () => {
    expect(Component).toMatchSnapshot();
  });
});
