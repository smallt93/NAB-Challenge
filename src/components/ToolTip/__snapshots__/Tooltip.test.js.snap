// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Tooltip component should match snapshot 1`] = `ReactWrapper {}`;
